cat-l1b.sh

```diff
@@ -1,7 +1,9 @@
 #!/bin/bash -u
 
 # keeping track of where I am
-DIR_NOW=$( cd $(dirname $0); pwd )
+DIR_NOW="$(cd $(dirname $BASH_SOURCE); pwd )"
```

Comment:
Possibly the change that will address the issue with blanks in the path is:

`DIR_NOW="$(cd "$(dirname $BASH_SOURCE)"; pwd )"`

Reply:
ASA we call $DIR_NOW somewhere else, the directory will break up again.

Reply:
Not sure what you mean, can you given an example or propose a solution?

Reply:
Just keep it as it is. But double quota anywhere we would use `DIR_NOW`， i.e. `"$DIR_NOW"`

Reply:
That's correct, otherwise any blank in the string $DIRNOW is interpreted by the shell as a new command. Quotes makes sure the shell doesn't do that and see the whole string in DIRNOW as a single string.